auth/src/main/java/com/firebase/ui/auth/viewmodel/idp/ProviderResponseHandlerBase.java

```diff
@@ -0,0 +1,50 @@
+package com.firebase.ui.auth.viewmodel.idp;
+
+import android.app.Application;
+import android.support.annotation.NonNull;
+import android.support.annotation.RestrictTo;
+
+import com.firebase.ui.auth.IdpResponse;
+import com.firebase.ui.auth.data.model.Resource;
+import com.firebase.ui.auth.util.data.ProviderUtils;
+import com.firebase.ui.auth.viewmodel.AuthViewModelBase;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.EmailAuthProvider;
+import com.google.firebase.auth.PhoneAuthProvider;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public abstract class ProviderResponseHandlerBase extends AuthViewModelBase<IdpResponse> {
```

Comment:
So the params to a viewmodel also contain another viewmodel?  That doesn't seem right to me (although I'm not an expert on arch components)

Reply:
Yup, fixed.